api/src/services/platform-service.ts

```diff
@@ -185,15 +199,31 @@ export class PlatformService extends DBService {
 
     dwcArchiveZip.addFile('eml.xml', Buffer.from(emlString));
 
-    const dwCADataset = {
+    const dwCADataset: IDwCADataset = {
       archiveFile: {
         data: dwcArchiveZip.toBuffer(),
-        fileName: 'DwCA.zip',
+        fileName: `${emlService.packageId}.zip`,
         mimeType: 'application/zip'
       },
-      dataPackageId: emlService.packageId
+      dataPackageId: emlService.packageId,
+      securityRequest
     };
 
-    return this._submitDwCADatasetToBioHubBackbone(dwCADataset);
+    const queueResponse = await this._submitDwCADatasetToBioHubBackbone(dwCADataset);
+
+    Promise.all([
```

Comment:
Should we put an `await` here?

Reply:
I don't think so, don't need an await for last call in a function

Reply:
> I don't think so, don't need an await for last call in a function

as long as you return the promise

Reply:
updated and added an await